testing/projects/cpp_nesting/__init__.py

```diff
@@ -9,6 +9,42 @@ def default_class_hierarchy_dict():
     """Return the default class hierarchy dictionary."""
     return {
         clike("struct", "top_level"): {},
+        namespace("special"): {
+            clike("struct", "Normal"): {
+                clike("struct", "Nested"): {
+                    clike("struct", "Like"): {
+                        clike("struct", "Usual"): {}
+                    }
+                }
+            },
+            clike("struct", "Image", template=["int width", "int height"]): {},
+            clike("class", "ImageBuffer", template=[
+                "class TImage", "typename Tdata_t", "int Tchannels"]): {
+                clike("struct", "Data"): {},
+                clike("struct", "SomeThing", template=["int Tnested"]): {}
+            },
+            clike("class", "ImageBuffer", template=[
+                "class TImage", "typename Tdata_t"]): {
+                clike("struct", "Data"): {},
+                clike("struct", "SomeThing"): {}
+            },
+            clike("class", "ImageBuffer"): {
+                clike("struct", "Data"): {},
+                clike("struct", "SomeThing"): {}
+            },
+            # TODO: undo #if 0 in nesting_specializations.hpp
+            # clike("struct", "Base", template=["size_t N"]): {
```

Comment:
If you uncomment this you have to uncomment the file hierarchy one below and undo `#if 0` in `include/nesting_specializations.hpp` and `src/tests.cpp` (to make sure it compiles, but dont worry about `tests.cpp` it is complicated to build but CI runs it)